### ISSUE TITLE
Using Nylas Node v3 SDK, create a code sample to update calendar event

### ISSUE THREAD

**Issue by relaxedtomato** [nylas-samples/nylas-hacktoberfest-2023#27]
Using Nylas Node v3 SDK, create a code sample to update calendar event

**Comment by destryptor** [comment]
Hello, I'd like to work on the issues with the node label, since I'm experienced in using node. Can you tell me how to structure the code?
Like for example, we need to initialise the Nylas Client with our API credentials and also use an access token, so should I just put placeholders like "YOUR_APPLICATION_ID" for those fields?

**Comment by relaxedtomato** [comment]
Thanks, @destryptor - just assigned this to you. Some further details:

1. Here are the details to install the Nylas SDK for Nylas v3: https://www.npmjs.com/package/nylas/v/7.0.0-beta.2
2. Send us an email at `devrel` @ `nylas.com` and we can help set up a dev account for testing the code sample, this should answer your inquiry above. I recommend testing out the code sample before creating a PR. I recently went over [how to create a code sample](https://github.com/nylas-samples/nylas-hacktoberfest-2023#youtube-livestream) in our last live stream to submit to Hacktoberfest.

**Comment by destryptor** [comment]
I went through the live stream, it was pretty helpful. I also took a look at the nylas hacktoberfest 2022 repo for additional help, and I have come up with a code for now, which is as belows:
![Screenshot 2023-10-08 173704](https://github.com/nylas-samples/nylas-hacktoberfest-2023/assets/117581936/3c62633a-b7c0-4fe6-8b64-8f3e342351a3)
In order to fetch the event by its ID, I have thought of including it in the .env file, which is as follows:
![image](https://github.com/nylas-samples/nylas-hacktoberfest-2023/assets/117581936/c83db4c0-c3d8-4833-aa27-10fb7f610b20)
The client will have to edit the fields like event.title, event.description etc. in index.js themselves to update the event.

It's my first time using the Nylas API, so I would be happy to know what else I need to add in the code, like Nylas libraries, more fields to update, or any other stuff in general. Also I'll send an email to devrel@nylas.com for the dev account to test the code out.
Thanks! Looking forward to hear from you soon.

**Comment by destryptor** [comment]
I went through the documentation, and I checked the model for Event. Is the calender event model also supposed to be the same?Then I think I need to update a few fields.

**Comment by relaxedtomato** [comment]
@destryptor  - got your message for an account, let me share that shortly. Will revisit your comments soon 👍

**Comment by destryptor** [comment]
Sure!

**Comment by relaxedtomato** [comment]
Sending it now :)

**Comment by relaxedtomato** [comment]
thanks for the email, next steps:

1. @destryptor accept verification email from Nylas
2. @relaxedtomato to set up dev dashboard for testing
3. @relaxedtomato to provide access to dev dashboard

**Comment by destryptor** [comment]
I have verified my email.

**Comment by relaxedtomato** [comment]
Steps 1, 2, and 3 are done :)

**Comment by destryptor** [comment]
I am not being able to reset my password, it keeps saying "Something unexpected happened. Please try again later"

**Comment by relaxedtomato** [comment]
@destryptor - were you able to access the dashboard?

**Comment by destryptor** [comment]
yes, I have access to it now. Thanks :) will be setting up the API key soon

**Comment by destryptor** [comment]
The only issue I faced was that the link in the reset password email (that was sent when you sent the request for password change) always returned "Some internal error occured, please try again later" no matter how many times I tried. However, when I requested to change the password myself on the v3 website, it worked.

**Comment by destryptor** [comment]
After testing, I found my program running fine. I have made the PR, do let me know if any changes are required. Thanks!

**Comment by relaxedtomato** [comment]
@destryptor - is this PR on your radar to prepare for this week? I've seen we've had a few comments, so I also created an faq for working with the correct dashboard.

https://github.com/nylas-samples/nylas-hacktoberfest-2023/blob/main/faqs.md

**Comment by destryptor** [comment]
> @destryptor - is this PR on your radar to prepare for this week? I've seen we've had a few comments, so I also created an faq for working with the correct dashboard.
> 
> https://github.com/nylas-samples/nylas-hacktoberfest-2023/blob/main/faqs.md

Please check my PR once, in one of the latest changes you requested (regarding incorrect params), I've put a comment asking for some help. Do let me know if you can view it.

**Comment by relaxedtomato** [comment]
@destryptor - I left PR comments, but didn't see your comment, can you link to it?

**Comment by destryptor** [comment]
> @destryptor - I left PR comments, but didn't see your comment, can you link to it?
> 
> https://github.com/nylas-samples/nylas-hacktoberfest-2023/pull/79#discussion_r1358833844
> 
> 

Yeah sure, it's:
https://github.com/nylas-samples/nylas-hacktoberfest-2023/pull/79#discussion_r1359404648

**Comment by relaxedtomato** [comment]
> > @destryptor - I left PR comments, but didn't see your comment, can you link to it?
> > https://github.com/nylas-samples/nylas-hacktoberfest-2023/pull/79#discussion_r1358833844
> 
> Yeah sure, it's: [#79 (comment)](https://github.com/nylas-samples/nylas-hacktoberfest-2023/pull/79#discussion_r1359404648)
> 
> Just after your last comment

@destryptor - The link just links to the PR, can you share a screen shot or add the comment here for reference? It's really hard to find things - did you test out the link?

I cannot find: https://github.com/nylas-samples/nylas-hacktoberfest-2023/pull/79#discussion_r1359404648

**Comment by relaxedtomato** [comment]
@destryptor here is an example of a comment with a link: https://github.com/nylas-samples/nylas-hacktoberfest-2023/pull/79/files#r1362152998

Are you able to access it?

**Comment by relaxedtomato** [comment]
Interestingly, searching for your comment is not possible 🤔 

https://github.com/search?q=repo%3Anylas-samples%2Fnylas-hacktoberfest-2023%20%20I%20checked%20the%20v3%20documentation%2C%20and%20I%20saw%20this%3A%20So%20the%20update%20even%E2%80%A6&type=code

**Comment by destryptor** [comment]
> Interestingly, searching for your comment is not possible 🤔 
> 
> https://github.com/search?q=repo%3Anylas-samples%2Fnylas-hacktoberfest-2023%20%20I%20checked%20the%20v3%20documentation%2C%20and%20I%20saw%20this%3A%20So%20the%20update%20even%E2%80%A6&type=code

I'll attach a screenshot of my comment:
![Screenshot_2023-10-17-22-55-54-211_com.github.android-edit.jpg](https://github.com/nylas-samples/nylas-hacktoberfest-2023/assets/117581936/cd7b645a-71a5-4e2e-abfa-8234ba750c8f)

Hope it's visible clearly.

**Comment by relaxedtomato** [comment]
Okay, let me respond here w/ reference numbers:

1. `identifier` is the correct key to use, so you need to provide a grant Id
2. when you receive an error, are you able to share the error? This is how we can advise further :)
3. when using the docs, you need to complete [this auth page first with API KEY](https://developer.nylas.com/docs/api/v3-beta/ecc/#auth) before any doc curl command will apply
4. I doubt it will be related to wifi, so it must be either the confirmation or the code, so having item 2 (error received) shared will help
5. Can you share your latest code as part of the PR so I can review the code in line and provide comments?

**Comment by relaxedtomato** [comment]
@destryptor - I like to help close this ticket, would you be interested in hopping on a code share to go over the code and progress the PR?

**Comment by destryptor** [comment]
> @destryptor - I like to help close this ticket, would you be interested in hopping on a code share to go over the code and progress the PR?

Thank you for your concern. I'll let you know what's going on right now. It's my bad that I didn't update you about this. Basically, my exams have started. So I am not having enough time to devote to this. The exams will end in 2 weeks. And by then, hacktoberfest will be over. I'm really sorry I couldn't match up my schedule. If you want, please feel free to assign this issue to someone else. I'll try to participate again next year. 
Thank you for your patience and all the help you provided, there were shortcomings from my side.

**Comment by relaxedtomato** [comment]
@destryptor - understood. Happy to help you complete the PR after exams end, let us know 👍 

Regardless of hacktoberfest ending in a week, it will be great to have this code sample merged :)
